service/src/main/java/com/wso2telco/dep/tpservice/pool/TokenReGenarator.java

```diff
@@ -98,21 +112,55 @@ public TokenDTO reGenarate(final WhoDTO who, final TokenDTO oldToken) throws Tok
 		// validate response message
 		if (Strtoken != null && Strtoken.length() > 0) {
 
-			JSONObject jsontoken = new JSONObject(Strtoken);
+			try {
+				jsontoken = new JSONObject(Strtoken);
```

Comment:
So here whenever a proper json response is not returned, retry will be triggered as the same same RESPONSE_CODE_ERROR is thrown.
Is this a requirement?